src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -382,6 +382,8 @@ class TeamUtilsCommonBase
 protected:
   int _team_size, _num_teams, _max_threads, _league_size;
 
+  TeamUtilsCommonBase() = default;
```

Comment:
Now that default ctor is legal, it is possible that someone forgets to call setup before using the class.

Do you think it's worth it to add some checks (maybe just in debug mode) every time get methods are called? Something like

```
int get_blah () const {
  EKAT_REQUIRE_MSG (_team_size>0, "Error! TeamUtils not yet inited.\n");
  return blah;
}
```
And also change `int _team_size` to `int _team_size=0,...`, to _guarantee_ that it's inited to 0 (and trigger the check error if no setup method is called).

Reply:
Yeah, I agree. I'll add checks.